File: LeetCodeJava/28.实现-str-str.java

/*
 * @lc app=leetcode.cn id=28 lang=java
 *
 * [28] 实现 strStr()
 */
class Solution {
    public int strStr(String haystack, String needle) {
        int len = needle.length();
        if (len == 0) {
            return 0;
        }
        String substr;
        int ret = -1;
        for (int i = 0; i < haystack.length() - len + 1; i ++) {
            substr = haystack.substring(i, i + len);
            if (substr.equals(needle)) {
                ret = i;
                break;
            }
        }
        return ret;
    }
}

File: LeetCodePython/28.实现-str-str.py
#
# @lc app=leetcode.cn id=28 lang=python3
#
# [28] 实现 strStr()
#

# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if haystack == "" and needle == "":
            return 0
        lneedle = len(needle)
        for i in range(len(haystack)):
            if haystack[i:i + lneedle] == needle:
                return i
        return -1

# @lc code=end

